collectors/plugins.d/plugins_d.c

```diff
@@ -154,7 +154,12 @@ inline size_t pluginsd_process(RRDHOST *host, struct plugind *cd, FILE *fp, int
 
         char *r = fgets(line, PLUGINSD_LINE_MAX, fp);
         if(unlikely(!r)) {
-            error("read failed");
+            if(feof(fp))
```

Comment:
How about some of those to get more fine grained error report for any error
```
NAME
     perror, strerror, strerror_r, sys_errlist, sys_nerr -- system error messages

LIBRARY
     Standard C Library (libc, -lc)

SYNOPSIS
     #include <stdio.h>

     void
     perror(const char *s);

     extern const char * const sys_errlist[];
     extern const int sys_nerr;

     #include <string.h>

     char *
     strerror(int errnum);

     int
     strerror_r(int errnum, char *strerrbuf, size_t buflen);
```

man perror should give you the details

Reply:
netdata `error` already does `strerror()` inside.